### ISSUE TITLE
Server returned invalid handshake?

### ISSUE THREAD

**Issue by gam3derp** [achimala/leaguelib#35]
Hello,
today lol have been updated to v4.2. from that time I'm getting this error while trying to use this lib: "Server returned invalid handshake". Any ideas why and how to fix this?

**Comment by andrebispo5** [comment]
Having the same problem here dunno what happen

**Comment by ghost** [comment]
Same problem here :/

2014-02-12 1:34 GMT+01:00 André Filipe da Silva Bispo <
notifications@github.com>:

**Comment by TheJasperV** [comment]
Same issue! Annoying that it's an rtmp issue.

**Comment by kzay** [comment]
Happy to see that i'm not alone ! but i'm not sure it's application issue. I have the same problem on my server but, im my personal computer in DEV environement it's work.

Difference between serv/dev , one on ubuntu other on windows. 
Maybe server ip ban by Riot?

If anyone have an idea. Thanks in advance.

**Comment by andrebispo5** [comment]
lolTeam is not working i guess its this problem :/

**Comment by gam3derp** [comment]
You are wrong. Its working

**Comment by andrebispo5** [comment]
no its not search a player

**Comment by achimala** [comment]
Hey guys. I'm not seeing this error on my end or on LoLTeam. What servers are you on? Could you post more specific steps to reproduce this error or a stack trace?

**Comment by gam3derp** [comment]
Hey, I've noticed this bug just after 4.2 update. every time i try to run the lib, i get this error: Server returned invalid handshake. I've tried on: eune, euw, na br. None worked, all gave the same error. Also, I've also tried to search for ongoing match just after I've noticed with bug via LoLTeam. it wasn't working for some time. but yesterday it started to work. I'm using windows 7, eclipse. Also tomcat 7.0 server isn't working. If need some more additional information, just let me know and i will try provide as much info as i can to get this fixed ;)

**Comment by kzay** [comment]
Im looking for a solution, but log dont give a lot of informations

**Comment by TheJasperV** [comment]
it's working again for me without code changes. Maybe the platform instability caused this in some way even though I could log in through league client but not leaguelib.

**Comment by gam3derp** [comment]
Its not working for me yet, I've got no idea why.  i also can  connect to server through client without any problems

**Comment by kzay** [comment]
i tried with 10account, some time one work, some time no.... 95% of time is invalid hanshake

**Comment by gam3derp** [comment]
i've also tried 3 accounts in 4 different servers: eune, euw, br and na. only euw works, other gives the same error. That is strange :/

**Comment by gam3derp** [comment]
hello guys, i wanna let you know, that it started to work on all servers. not sure what that happened...

**Comment by farkerhaiku** [comment]
This started up again after the 4.17 patch.   Any idea if this is in preparation for the October 1st deadline to remove this stuff from your app?

**Comment by jaspervdm** [comment]
It still works fine for me. Did you set version to 4.17.xx?

**Comment by farkerhaiku** [comment]
Yeah, the problem resolved itself after several hours. It looks like it was
down for at least 5 hours.

**Comment by desmond-silveira** [comment]
4.17.xx and it's happening for me.  I assume that it's Riot-side, so I guess I'll just wait.

**Comment by desmond-silveira** [comment]
Error is gone now with no code change.

**Comment by ghost** [comment]
It's happening again and it's been like that since friday (or thursday, I can't even remember). Is it possible that riot have changed something on their servers?

**Comment by DQKev** [comment]
This are the Server data in LoLRtmpsClient.java
this.server = "prod.eu.lol.riotgames.com";
this.loginQueue = "https://lq.eu.lol.riotgames.com/";

This are working on my version ;)
"prod.euw1.lol.riotgames.com", "https://lq.euw1.lol.riotgames.com/"

Riot have changed some endpoints ;)

Take a look into my merge request:
https://github.com/achimala/leaguelib/pull/39/files

**Comment by desmond-silveira** [comment]
This started happening again for me.  It started when 4.20 was released and it has been going on for about a week now, but this time, the problem is only on my production server.  On my dev environment, everything seems to work fine.  I tried changing endpoints from

```
prod.na1.lol.riotgames.com
https://lq.na1.lol.riotgames.com/
```

to 

```
prod.na.lol.riotgames.com
https://lq.na.lol.riotgames.com/
```

or

```
prod.na2.lol.riotgames.com
https://lq.na2.lol.riotgames.com/
```

That didn't resolve the issue at all.  Even with the changes, it works on my dev but not in production.

The exception being thrown is `IOException("Server returned invalid handshake")` from `RTMPSClient.doHandshake()` so could it be that the root certificates with the OS are out of date or something like that?

**Comment by farkerhaiku** [comment]
I had the same problem desmond, and it was because my mac had oracle jdk, while my server had openjdk.  I'm kind of assuming that it had something to do with the JCE jars or whatever.  I'd try that and see if that works. Also, be sure to update to prod.na2

**Comment by jaspervdm** [comment]
I have added a pull request that fixes this issue, see: https://github.com/achimala/leaguelib/pull/44
